src/plugins/newsfeed/public/lib/storage.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+/**
+ * Persistence layer for the newsfeed driver
+ */
+export class NewsfeedStorage {
+  private readonly lastFetchStorageKey: string;
+  private readonly readStatusStorageKey: string;
+  private readonly unreadStatus$: BehaviorSubject<boolean>;
+
+  constructor(storagePrefix: string) {
+    this.lastFetchStorageKey = getStorageKey(storagePrefix, 'lastFetch');
+    this.readStatusStorageKey = getStorageKey(storagePrefix, 'readStatus');
+    this.unreadStatus$ = new BehaviorSubject<boolean>(anyUnread(this.getReadStatus()));
+  }
+
+  getLastFetchTime(): Date | undefined {
+    const lastFetchUtc = localStorage.getItem(this.lastFetchStorageKey);
+    if (!lastFetchUtc) {
+      return undefined;
+    }
+
+    return moment(lastFetchUtc, 'x').toDate(); // parse as unix ms timestamp (already is UTC)
+  }
+
+  setLastFetchTime(date: Date) {
+    localStorage.setItem(this.lastFetchStorageKey, JSON.stringify(date.getTime()));
+  }
+
+  setFetchedItems(itemHashes: string[]): boolean {
+    const currentReadStatus = this.getReadStatus();
+
+    const newReadStatus: Record<string, boolean> = {};
+    itemHashes.forEach((hash) => {
+      newReadStatus[hash] = currentReadStatus[hash] ?? false;
+    });
+
+    this.setReadStatus(newReadStatus);
+    return anyUnread(newReadStatus);
+  }
+
+  /**
+   * Marks given items as read, and return the overall unread status.
+   */
+  markItemsAsRead(itemHashes: string[]): boolean {
+    const updatedReadStatus = this.getReadStatus();
+    itemHashes.forEach((hash) => {
+      updatedReadStatus[hash] = true;
+    });
+    this.setReadStatus(updatedReadStatus);
+    return anyUnread(updatedReadStatus);
```

Comment:
Super-NIT: We are calling this method twice in a row (once inside `this.setReadStatus`, and then in this line). Should we aim to reuse the value somehow to avoid extra loops?